File: src/solutions/_012.java
package solutions;

public class _012 {

    public static long highlyDivisibleTriangularNumber() {
        int num = 0, i = 1;
        long number = 0;
        while(num < 500) {
            number = (i*(i + 1))/2;
            num = numberOfDivisors(number);
            i++;
        }
        return number;
    }

    private static int numberOfDivisors(long n) {
        int i;
        int num = 1;
        for(i = 2; i < Math.sqrt(n); i++) {
            if(n%i == 0) {
                num = num + 1;
            }
        }
        return num*2;
    }
}

File: c/src/012.c
#include "../include/012.h"
#include <math.h>

static int ndiv(int);

int highly_divisible_triangular_number(void)
{
    int i = 1;
    int n = 0;
    int r;

    while (n < 500)
    {
        r = (i * (i + 1)) / 2;
        n = ndiv(r);
        ++i;
    }

    return r;
}

int ndiv(int n)
{
    int i = 0;
    int d = 1;

    for (i = 2; i < sqrt(n); i++)
    {
        if (0 == n % i)
        {
            ++d;
        }
    }

    return 2 * d;
}
